internal/resolver/resolve.go

```diff
@@ -60,41 +151,28 @@ type varRef struct {
 	funcName string
 	ref      *ast.VarExpr
 	isArg    bool
-	pos      Position
 }
 
 // A single array reference
 type arrayRef struct {
 	funcName string
 	ref      *ast.ArrayExpr
-	pos      Position
 }
 
 // Initialize the resolver
-func (p *parser) initResolve() {
-	p.varTypes = make(map[string]map[string]typeInfo)
-	p.varTypes[""] = make(map[string]typeInfo) // globals
-	p.functions = make(map[string]int)
-	p.arrayRef("ARGV", Position{1, 1})    // interpreter relies on ARGV being present
-	p.arrayRef("ENVIRON", Position{1, 1}) // and other built-in arrays
-	p.arrayRef("FIELDS", Position{1, 1})
-	p.multiExprs = make(map[*ast.MultiExpr]Position, 3)
-}
-
-// Signal the start of a function
-func (p *parser) startFunction(name string, params []string) {
-	p.funcName = name
-	p.varTypes[name] = make(map[string]typeInfo)
-}
-
-// Signal the end of a function
-func (p *parser) stopFunction() {
-	p.funcName = ""
-}
-
-// Add function by name with given index
-func (p *parser) addFunction(name string, index int) {
-	p.functions[name] = index
+func (r *resolver) initResolve(config *Config) {
```

Comment:
Nit: this could be named just `init` now that it's on the `resolver` type. In fact, what about making it a `newResolver(config *Config)` function instead?